### DISCUSSION TITLE
Using SqlConnections.NewFor<XXX>() with new .NET 5 serene

### DISCUSSION THREAD

**Discussion by JohnRanger** [serenity-is/Serenity#5443] [Help]
Hi all,

until recently, I easily could access the DB from within self-contained CS functions by creating a connection like this:

```cs
    var Connection = Serenity.Data.SqlConnections.NewFor<myLanguageRow>();
```
and then build upon this.

Now with the .NET5 version of Serenity framework, SqlConnections is not anymore available within Serenity.Data.
I have seen that there is ISqlConnections - but I couldn't figure out how to use it (I am no real programmer, just an IT admin).

As ```ISqlConnections sqlConnections = new ISqlConnections();```doesn't work.

Any help how to "in-place" replace ```var Connection = Serenity.Data.SqlConnections.NewFor<myLanguageRow>();``` is very much appreciated.

**The goal**
...is to not having to pass the connection object to these helper functions - neither by parameter nor via interface nor via constructor.

Function library looks like so:

```cs
public static class LanguageHelpers
    {
        public static int getCurrentUIlanguageID()
        {
            var UI_languageShortName = CultureInfo.CurrentUICulture.Name.Split('-')[0]; // *** the split handles cases where a subset of the main language is detected - e.g. en-CH ***

            var _myLanguageRow = new myLanguageRow();

            var Connection = sqlConnections.NewFor<myLanguageRow>();      // <<<--- here Serenity.Data.sqlconnections doesn't exist anymore
            .
            .
            .

```

With kind regards,

John

**Comment by erehtolleh** [comment]
He abstract them now. You have to put it in your class constructor or pass the connection through the methods. See the migration 3-5 document.  
https://serenity.is/Docs/migration/v3-to-v5

You can look at the EmailSender.cs or Backgroundjob class

You have to do something like below:

```
public class MailingBackgroundJob : PeriodicBackgroundJob
{
    private readonly ISqlConnections connections;
    private readonly IEmailSender emailSender;
    private readonly MailingServiceSettings config;
    private readonly IExceptionLogger logger;

    public MailingBackgroundJob(ISqlConnections connections, IEmailSender emailSender, 
        IOptions<MailingServiceSettings> options, IExceptionLogger logger = null)
    {
        this.connections = connections ?? throw new ArgumentNullException(nameof(connections));
        this.emailSender = emailSender ?? throw new ArgumentNullException(nameof(emailSender));
        config = (options ?? throw new ArgumentNullException(nameof(options))).Value;
        this.logger = logger;
    }
```

**Reply by JohnRanger** [reply]
Hi @erehtolleh ,

thanks for your feedback. 

 I have seen such as these - but as I want to encapsulate my generic helper methods as best as possible, I want to open the necessary connections from within my helper methods - so they work also in contexts where no connection is available.

If I understand the example given by you correctly, I still would have to provide the connection from the outside of the class - by filling in the relevant parameter when instantiating the class. This is not what I want.

Hence my question :-)

With kind regards,

John

**Comment by erehtolleh** [comment]
I see what you are saying, didn't read your "goal" line there.  I recently had to write the static class but just passed it in via parameters.  I would be interested in this also if you find an answer.

**Reply by JohnRanger** [reply]
@erehtolleh ,

thanks by the way for the link to the 3-to-5 migration guide. This one I had previously not known. 

With kind regards,

John

**Reply by erehtolleh** [reply]
Thanks for some of the useful Wiki...I appreciated it.

**Reply by WasimAhmad** [reply]
hi @JohnRanger you can create a service and inject it before using..
e.g

```csharp
namespace xxx.Services
{
    public interface ILanguageHelpers
    {
        int getCurrentUIlanguageID();
    }
}

//Impemenation Class

namespace xxx.Services
{
    public class LanguageHelpers : ILanguageHelpers
    {
        public LanguageHelpers(ISqlConnections sqlConnections)
        {
            SqlConnections = sqlConnections;
        }

        public ISqlConnections SqlConnections { get; }

        public int getCurrentUIlanguageID()
        {
            var UI_languageShortName = CultureInfo.CurrentUICulture.Name.Split('-')[0]; // *** the split handles cases where a subset of the main language is detected - e.g. en-CH ***

            var _myLanguageRow = new myLanguageRow();

            var Connection = SqlConnections.NewFor<myLanguageRow>();      // <<<--- here Serenity.Data.sqlconnections doesn't exist anymore
        }
    }
}


// add into DI pipeline 

services.AddSingleton<ILanguageHelpers, LanguageHelpers>();

**Reply by JohnRanger** [reply]
Hi @WasimAhmad ,
thanks - will try that and post result here :-)

With kind regards,

John
